src/avisi/atlassian/connect/interceptors.clj

```diff
@@ -56,6 +57,49 @@
                   {:enter (fn [ctx]
                             (update ctx :request host-request options))})))})
 
+(defn invalid-license? [{:keys [query-params]}] (not= (:lic query-params) "active"))
+
+(defn atlassian-marketplace-license-ex-handler [invalid-license-html]
+  (fn
+    [{:keys [request error]
+      :as ctx}]
+    (if (= :invalid-license (:type (ex-data error)))
+      (do
+        (log/warn
+          {:request-method (:request-method request)
+           :query-string (:query-string request)
+           :uri (:uri request)}
+          error
+          "Invalid license detected")
+        (->
+          ctx
+          (dissoc :error)
+          (assoc
+            :response
+            (->
+              invalid-license-html
+              response/response
+              (response/status 403)
+              (response/content-type "text/html")
+              (response/header "Content-Security-Policy" "")))))
+      ctx)))
+
+(defn check-license? [uri skip-license-check?]
+  (and (not skip-license-check?) (not= uri "/connect/atlassian-connect.json")))
+
+(defn validate-license-interceptor [{:keys [skip-license-check? invalid-license-html]}]
+  {:name ::validate-license-interceptor
+   :error (atlassian-marketplace-license-ex-handler invalid-license-html)
+   :compile
+   (fn [_ _]
+     {:enter
+      (fn
+        [{:keys [request]
+          :as ctx}]
+        (if (and (check-license? (:uri request) skip-license-check?) (invalid-license? request))
+          (throw (ex-info "Invalid license" {:type :invalid-license}))
```

Comment:
Ik zou geen exceptie gebruiken maar gewoon de invalid license pagina returnen in plaats van eerst een exceptie te gooien en deze verderop weer te gebruiken.